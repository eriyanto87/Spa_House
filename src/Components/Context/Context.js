import React from "react";

const Context = React.createContext({
  // users: [],
  // cart: [],
  // treatments: [],
  // addUser: () => {},
  // addCart: () => {},
  //   getAllTreatments: () => {},
});

export default Context;
