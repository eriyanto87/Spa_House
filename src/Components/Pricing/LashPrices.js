import React from "react";
import { Link } from "react-router-dom";

export default function LashPrices() {
  return (
    <div>
      <section>
        <h3>Lash Prices</h3>
        <h3>Classic </h3>
        <p>
          <input type="radio" id="fullset" value="79" />{" "}
          <label for="one-hour">Full Set: $99</label>
        </p>
        <input type="radio" id="fill" value="59" />
        <label for="one-hour">Refill: $79</label>
        <br />
        <button type="cancel">Cancel</button>
        <Link to="/schedule">
          <button type="submit">Schedule</button>
        </Link>
      </section>
    </div>
  );
}
