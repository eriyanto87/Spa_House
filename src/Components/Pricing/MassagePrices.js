import React from "react-router-dom";
import { Link } from "react-router-dom";

export default function MassagePrices() {
  return (
    <div>
      <section>
        <h3>Massage Pricing</h3>
        <input type="radio" id="60" value="99" />
        <label for="one-hour">60 minutes: $99</label>
        <input type="radio" id="90" value="139" />
        <label for="one-hour">90 minutes: $139</label>
        <br />
        <button type="cancel">Back</button>
        <Link to="/schedule">
          <button>Schedule</button>
        </Link>
      </section>
    </div>
  );
}
