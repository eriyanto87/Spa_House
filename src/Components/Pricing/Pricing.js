import React from "react";
import Context from "../Context/Context";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";

export default function MassagePrices(props) {
  const goBack = () => props.history.goBack();

  const onSubmit = (e, context) => {
    e.preventDefault();
    console.log(context.treatment);
    if (context.treatment > 0) {
      props.history.push("/schedule");
    } else {
      context.setError("Please choose a service");
    }
  };

  const getTreatmentId = (e, context) => {
    const treatmentId = e.target.id;
    console.log(treatmentId);
    context.setTreatment(treatmentId);
    console.log(context.treatment);
  };

  return (
    <Context.Consumer>
      {(context) => {
        return (
          <div className="pricing">
            <Nav />
            <section className="pricingSection">
              <h3>PRICING</h3>
              <p className="pricingHighlight">{context.error}</p>
              <form
                onSubmit={(e) => onSubmit(e, context)}
                onChange={(e) => getTreatmentId(e, context)}
              >
                {context.treatments.map((t, i) => (
                  <p className="pricingContainer" key={i}>
                    <label>
                      <input
                        className="price"
                        type="radio"
                        id={t.id}
                        name="price"
                        value={context.name}
                        checked={t.id == context.treatment}
                      />
                      {t.display_name}-{" "}
                      <span className="pricingHighlight">${t.price}</span>
                    </label>
                  </p>
                ))}
                <button onClick={goBack} type="cancel">
                  BACK
                </button>{" "}
                <button type="submit">SCHEDULE</button>
              </form>
            </section>
            <Footer />
          </div>
        );
      }}
    </Context.Consumer>
  );
}
