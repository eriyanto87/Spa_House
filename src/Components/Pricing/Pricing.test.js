import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Context from "../Context/Context";
import Pricing from "./Pricing";

describe("Pricing component", () => {
  const contextValues = {
    setError: () => {},
    setTreatment: () => {},
    getTreatment: () => {},
    treatments: [],
  };

  it("Pricing Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Context.Provider value={contextValues}>
            <Pricing />
          </Context.Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={contextValues}>
          <Pricing />
        </Context.Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
