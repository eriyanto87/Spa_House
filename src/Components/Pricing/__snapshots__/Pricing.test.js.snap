// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing component Pricing Snapshot 1`] = `
<div
  className="pricing"
>
  <div
    className="nav"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About 
        </a>
      </li>
      <li>
        <a
          href="/treatments"
          onClick={[Function]}
        >
          Treatments 
        </a>
      </li>
      <li>
        <a
          href="/pricing"
          onClick={[Function]}
        >
          Pricing 
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact Us 
        </a>
      </li>
      <li>
        <a
          href="/gallery"
          onClick={[Function]}
        >
          Gallery
        </a>
      </li>
    </ul>
  </div>
  <section
    className="pricingSection"
  >
    <h3>
      PRICING
    </h3>
    <p
      className="pricingHighlight"
    />
    <form
      onChange={[Function]}
      onSubmit={[Function]}
    >
      <button
        onClick={[Function]}
        type="cancel"
      >
        BACK
      </button>
       
      <button
        type="submit"
      >
        SCHEDULE
      </button>
    </form>
  </section>
  <div>
    <p>
      Â© 2020
    </p>
  </div>
</div>
`;
